webapp/sources/rudder/rudder-core/src/main/scala/com/normation/plugins/PluginSettings.scala

```diff
@@ -273,3 +281,259 @@ class FilePluginSettingsService(pluginConfFile: File, readSetupDone: IOResult[Bo
+        .withFieldConst(_.softwareId, softwareId.value)
+        .withFieldConst(_.minVersion, minVersion.value)
+        .withFieldConst(_.maxVersion, maxVersion.value)
+        .withFieldConst(_.maxNodes, maxNodes.value)
+        .withFieldConst(_.others, Map.empty[String, String])
+        .buildTransformer
+    }
+  }
+
+  implicit val decoder: JsonDecoder[RudderPackagePlugin] = DeriveJsonDecoder.gen[RudderPackagePlugin]
+
+  /**
+    * When joining plugin information from rudder package and global information from registered plugins,
+    * we can return needed plugin details
+    */
+  implicit def transformer(implicit
+      rudderFullVersion: String,
+      abiVersion:        Version,
+      transformLicense:  Transformer[LicenseInfo, PluginLicenseInfo]
+  ): Transformer[RudderPackagePlugin, JsonPluginSystemDetails] = {
+    val _ = transformLicense // variable is used below
+    Transformer
+      .define[RudderPackagePlugin, JsonPluginSystemDetails]
+      .withFieldComputed(_.id, _.name)
+      .withFieldComputed(
+        _.status,
+        p => {
+          (p.installed, p.enabled) match {
+            case (true, true)  => JsonPluginSystemStatus.Enabled
+            case (true, false) => JsonPluginSystemStatus.Disabled
+            case (false, _)    => JsonPluginSystemStatus.Uninstalled
+          }
+        }
+      )
+      .withFieldComputed(_.version, l => l.version.getOrElse(l.latestVersion))
+      .withFieldConst(_.abiVersion, abiVersion) // field is computed upstream
+      .withFieldComputed(_.pluginType, p => if (p.webappPlugin) PluginType.Webapp else PluginType.Integration)
+      .withFieldConst(_.statusMessage, None)
+      .withFieldComputed(
+        _.errors,
+        PluginManagementError.fromRudderPackagePlugin(_).map(_.transformInto[JsonPluginManagementError])
+      )
+      .buildTransformer
+  }
+}
+
+/**
+  * A service that encapsulate rudder package operations and its representation of plugins.
+  */
+trait RudderPackageService {
+  def listAllPlugins(): IOResult[Chunk[RudderPackagePlugin]]
+
+  def installPlugins(plugins: Chunk[String]): IOResult[Unit]
+
+  def removePlugins(plugins:           Chunk[String]): IOResult[Unit]
+  def changePluginSystemStatus(status: PluginSystemStatus, plugins: Chunk[String]): IOResult[Unit]
+}
+
+class RudderPackageCmdService extends RudderPackageService {
+
+  private val listCmd = Cmd("rudder", "package" :: "list" :: "--all" :: "--format=json" :: Nil, Map.empty, None)
+
+  override def listAllPlugins(): IOResult[Chunk[RudderPackagePlugin]] = {
+    val cmd = listCmd
+    for {
+      packagesCmd <- RunNuCommand.run(cmd)
+      result      <- packagesCmd.await
+      _           <- ZIO.when(result.code != 0) {
+                       Inconsistency(
+                         s"An error occurred while listing packages with '${cmd.display}':\n code: ${result.code}\n stderr: ${result.stderr}\n stdout: ${result.stdout}"
+                       ).fail
```

Comment:
we need to log that in console too, because it will need to be available for terminal work and debuging.